### PR TITLE
🔀[build] Port build script to windows

### PR DESCRIPTION
**Pull request by Cr4zyc4k3**
**What kind of update does this PR provide?** *Please check*

- [ ] new translations / updated translations / translation fixes
- [ ] a new feature
- [ ] a new module
- [ ] a bugfix for an existing feature / module
- [x] improvement of style or user experience
- [ ] other: *Please fill out*

**What is included in your update?**
* Port [build/build.sh](https://github.com/LSS-Manager/LSSM-V.4/blob/dev/build/build.sh) to windows powershell script

### CODE REVIEW

**Comment by Cr4zyc4k3** [comment]
@deoxis9001 At least on my installation it works with ```cmd
.\build\build.ps1 --local
```

**Comment by Cr4zyc4k3** [comment]
Did you execute `yarn install` before?

**Comment by deoxis9001** [comment]
yes I did it and it didn't change anything
I'm on windows 10 if that helps

**Comment by Cr4zyc4k3** [comment]
I think there are some issues with the PATH variable. I will inspect that.

**Comment by Cr4zyc4k3** [comment]
image
Seems like webstorm adds some stuff to the $PATH variable :D

**Comment by Cr4zyc4k3** [comment]
@deoxis9001 It now works on powershell in a default powershell on my system 🙈 
Fyi: Currently the `--port` command is not working so you need to adjust it in the script if you want to use another port than 36551

**Comment by deoxis9001** [comment]
image
image
is it possible to delete these messages locally?

**Comment by Cr4zyc4k3** [comment]
Probably yes, but the effort would be much higher than the result.

The quick&dirty solution:
For the branches should be an empty `branches.json` file in the dist folder enough to remove this error. 
For the release notes is a deeper inspection required.

**Comment by jxn-30** [comment]
> Probably yes, but the effort would be much higher than the result.
> 
> The quick&dirty solution: For the branches should be an empty `branches.json` file in the dist folder enough to remove this error. For the release notes is a deeper inspection required.
> 
> @jxn-30 Thoughts on this topic?

empty `branches.json` seems fine to me, maybe we could also provide empty Releasenote-Files or otherwise download them similar to the mission files in the prebuild step.

File: docs/en_GB/contributing/runningLocally.md
Comment (jxn-30) [inline docs/en_GB/contributing/runningLocally.md:1]: not necessary anymore since using symlink for en_GB

File: docs/en_GB/contributing~dev
Comment (jxn-30) [inline docs/en_GB/contributing~dev:1]: please undo symlink rename ;)

File: docs/en_US/contributing/committing.md [hunk 1]
```diff
@@ -22,18 +22,21 @@ what you changed (eg. translations). Some long module names got unofficial abbre
...
 | docs         | docs (main pages only)    |                         |
 | etr          | enhancedTransportRequests |                         |
-| hk           | hotkeys                   |                         |    
+| hk           | hotkeys                   |                         |
 
 ![](../images/contributing/committing/GH_commit_msg.png)
 
 :::tip What's next?
 
+* Read [Test locally](./runningLocally.md)
```
Comment (jxn-30) [inline docs/en_US/contributing/committing.md:39]: link to `./development` with fitting anchor?

File: docs/en_US/contributing/runningLocally.md
Comment (jxn-30) [inline docs/en_US/contributing/runningLocally.md:1]: how about integrating that into `./development`? I think that could reduce redundant information

File: scripts/buildUserscript.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import packageJson from '../package.json';
 import config, { PORT_ENV_KEY } from '../src/config';
 
 const script = packageJson.userscript;
+const localCoreName = 'core';
```
Comment (jxn-30) [inline scripts/buildUserscript.ts:11]: I think, we also have to pass this as parameter to Terser

**Comment by deoxis9001** [comment]
image
I have other errors

**Comment by kdev** [comment]
[Bun introduced](https://bun.sh/blog/the-bun-shell) a cross-os shell layer to.
Maybe it's worth a thought rewriting the build script in JS/TS with some sort of compile layer to make it run on multiple OS.
This would require a complete rewrite of the build script but would help maintainability a lot (something this project needs!).
Any thoughts?

**Comment by jxn-30** [comment]
> [Bun introduced](https://bun.sh/blog/the-bun-shell) a cross-os shell layer to. Maybe it's worth a thought rewriting the build script in JS/TS with some sort of compile layer to make it run on multiple OS. This would require a complete rewrite of the build script but would help maintainability a lot (something this project needs!). Any thoughts?
> 
> Edit: Bun is not the first to introduce something like this. [Google open-sourced a similar project years ago.](https://google.github.io/zx/)

I like the idea. That may also be useful for #2884 where currently weird child processes are spawned instead.

**Comment by Cr4zyc4k3** [comment]
Since google is famous for throwing project under the bus (https://killedbygoogle.com/) I won't relay on something they maintain. But I like the idea in general. Until this is done we should keep both versions.

**Comment by kdev** [comment]
> Since google is famous for throwing project under the bus (https://killedbygoogle.com/) I won't relay on something they maintain. But I like the idea in general. Until this is done we should keep both versions.

Luckily they state that it's not a "supported Google product". 
I guess they won't throw it under the bus since they are probably using it themselves (also the project is at v7 and has been around for a while now)
Anyways. The Google's zx is made for just linux scripts so it won't fit our use case.

@Cr4zyc4k3 what do you think about the Bun project?

**Comment by Cr4zyc4k3** [comment]
If it is possible to accomplish this without installing bun globally I'm cool with this solution.

File: docs/en_US/contributing/runningLocally.md
Comment (Cr4zyc4k3) [reply docs/en_US/contributing/runningLocally.md:1]: Cleanup in a seperate branch

**Comment by Cr4zyc4k3** [comment]
> image image is it possible to delete these messages locally?

Should be fixed with https://github.com/LSS-Manager/LSSM-V.4/commit/16c51fd59111ab0fb72d702c26441d60722f918f

File: scripts/buildUserscript.ts (hunk 2 above)
Comment (Cr4zyc4k3) [reply scripts/buildUserscript.ts:11]: Fixed in [d2c9831](https://github.com/LSS-Manager/LSSM-V.4/pull/2865/commits/d2c98310f1fb6c4e28c81efd0d0fda5a91e8487a)

**Review by jxn-30 — APPROVED** [review]

**Comment by kdev** [comment]
Bun now supports windows. Maybe we should consider switching/rewriting for better maintainability?

**Comment by jxn-30** [comment]
I think at that point we should take the chance and have a look at the whole CI and see where we could improve it.

**Pull request merged**